### PR TITLE
clean up build tests and make them run on 4.0 installer clusters

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@adambkaplan iterating on aws-builds here.

**Comment by bparees** [comment]
@smarterclayton i was working on getting all the build tests working on AWS here.  There are at least a few tests that need specific updates because of the change from "docker-registry" to "image-registry" when we go from ansible installed registries to operator managed ones.

**Comment by bparees** [comment]
/test e2e-aws-builds

@smarterclayton notice i added a new aws-builds job to eventually replace gcp-builds, it's currently optional until i sort out the failures.  

I also already moved gcp-image-ecosystem to aws-image-ecosystem, that one is passing already.

I suspect gcp-image-registry will take more work but i haven't looked at it yet.

**Comment by adambkaplan** [comment]
/lgtm

Green-lighting so we can get the build tests off of GCP. Noting here that there are two items we need to address:

1. Prometheus not scraping build metrics.
2. Need admin permissions to tear down jenkins template.

**Comment by bparees** [comment]
/test e2e-aws-builds

(would like to see another couple successes here..doesn't matter for merging this PR, but will matter before I switch the job over)

**Comment by bparees** [comment]
/hold

need to sort out why gcp-builds is skipping everything.

**Comment by bparees** [comment]
ok i think i sorted that out, let's see what the next set of build runs show.

**Comment by bparees** [comment]
ok gcp did what it should, aws-build looks like a networking flake bit it, but will rerun:

/retest
/test e2e-aws-builds

File: test/extended/prometheus/prometheus_builds.go [hunk 1]
```diff
@@ -38,11 +38,16 @@ var _ = g.Describe("[Feature:Prometheus][Feature:Builds] Prometheus", func() {
 
 	g.Describe("when installed on the cluster", func() {
 		g.It("should start and expose a secured proxy and verify build metrics", func() {
+			if exutil.IsClusterOperated(oc) {
+				// TODO: prometheus is not scraping build metrics on 4.0 installer clusters... something for the prometheus team
+				// to dig into?
```
Comment (smarterclayton) [inline test/extended/prometheus/prometheus_builds.go]: Please make sure there is an open bug here.

File: test/extended/builds/pipeline_origin_bld.go [hunk 2]
```diff
@@ -260,7 +260,16 @@ var _ = g.Describe("[Feature:Builds][Slow] openshift pipeline build", func() {
...
+				// service "mongodb" deleted
+				// deploymentconfig.apps.openshift.io "mongodb" deleted
+				// Error from server (Forbidden): clusterserviceversions.operators.coreos.com is forbidden: User "e2e-test-jenkins-pipeline-2z82q-user" cannot list clusterserviceversions.operators.coreos.com in the namespace "e2e-test-jenkins-pipeline-2z82q": no RBAC policy matched
+				// Error from server (Forbidden): catalogsources.operators.coreos.com is forbidden: User "e2e-test-jenkins-pipeline-2z82q-user" cannot list catalogsources.operators.coreos.com in the namespace "e2e-test-jenkins-pipeline-2z82q": no RBAC policy matched
+				// Error from server (Forbidden): installplans.operators.coreos.com is forbidden: User "e2e-test-jenkins-pipeline-2z82q-user" cannot list installplans.operators.coreos.com in the namespace "e2e-test-jenkins-pipeline-2z82q": no RBAC policy matched
+				// Error from server (Forbidden): subscriptions.operators.coreos.com is forbidden: User "e2e-test-jenkins-pipeline-2z82q-user" cannot list subscriptions.operators.coreos.com in the namespace "e2e-test-jenkins-pipeline-2z82q": no RBAC policy matched
```
Comment (smarterclayton) [inline test/extended/builds/pipeline_origin_bld.go]: I'm really confused.  Why would those errors be happening?

Comment (smarterclayton) [reply test/extended/builds/pipeline_origin_bld.go]: Have we not added those roles correctly to a project editor?

Comment (bparees) [reply test/extended/builds/pipeline_origin_bld.go]: I opened a bug against the OLM team about this.  QE also ran into it.  Either the roles aren't being added properly, or the OLM resources should not be part of the "all" alias.  Maciej also sent a note about this to aos-devel in the last couple days.

File: test/extended/prometheus/prometheus_builds.go (hunk 1 above)
Comment (bparees) [reply test/extended/prometheus/prometheus_builds.go]: https://jira.coreos.com/browse/DEVEXP-235

File: test/extended/builds/pipeline_origin_bld.go (hunk 2 above)
Comment (smarterclayton) [reply test/extended/builds/pipeline_origin_bld.go]: so you're working around someone else's bug

Comment (bparees) [reply test/extended/builds/pipeline_origin_bld.go]: correct.

**Pull request merged**